### PR TITLE
Pairing

### PR DESCRIPTION
**Pull request by steve-at-hx**
Guidance page on pair programming techniques and benefits.

### CODE REVIEW

File: pairing.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+## When should we pair?
+
+There's no concrete right or wrong time, but it's worth keeping in mind that the way you pair needs modifying to make sure you're achieving that goal. At HX, we would typically use pairing to onboard new hires, solve difficult problems and spread knowledge. 
+
+The Scrum Alliance describe the pair as a Driver, in control of the development environment and a Navigator providing instruction. This slightly over simplifies the relationship, so we'll break out some of the different ways we pair depending on the situation.
+
+### Onboarding
+
+During onboarding you would typically pair a new hire with an experienced engineer. The intention is for the experienced engineer to provide the context and safety net needed for the new hire to learn on a real platform. It has been proven, externally and internally, to be a really effective, supportive way of getting a new hire up to speed.
```
Comment (lizpope) [inline pairing.md]: Could you replace 'new hire' with 'new starter' throughout? (not a big deal, but 'starter' sounds more like a HX term than 'hire' does)

File: pairing.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+The Scrum Alliance describe the pair as a Driver, in control of the development environment and a Navigator providing instruction. This slightly over simplifies the relationship, so we'll break out some of the different ways we pair depending on the situation.
+
+### Onboarding
+
+During onboarding you would typically pair a new starter with an experienced engineer. The intention is for the experienced engineer to provide the context and safety net needed for the new starter to learn on a real platform. It has been proven, externally and internally, to be a really effective, supportive way of getting a new starter up to speed.
+
+This is the prime example where Driver and Navigator is an incomplete analogy. If the new starter is following instruction and doesn't understand *why* they're performing actions, they're unlikey to be able to complete a similar task independently in the future, which should be a key aim of pairing for onboarding. In pairing for onboarding the Navigator should more resemble a Coach.
```
Comment (jamesmcneil) [inline pairing.md]: Small typo here `unlikey` -> `unlikely`

File: pairing.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Now imagine you had a local in the car who was providing context and maybe giving you a little test along the way. Instead of saying "turn left", they might say "you'll need to turn left after this pub, which has a great garden by the way" and you may ask "and if I overshoot the turning", and they may say "don't worry you can take the next left as they join up". You hopefully get the drift, and would agree you'd be much more likely to be able to repeat the journey. 
+
+An effective onboarding pair will typically have the new starter at the keyboard with the experienced engineer providing a mix of instruction and questioning. It's this questioning, and opportunity for the new starter to question their own understanding, that helps the new starter to build the context around the instruction. There is some [theory](https://en.wikipedia.org/wiki/Action_learning#Revans.27_formula) that supports this style of learning.
```
Comment (jamesmcneil) [inline pairing.md:21]: We've used `new starter` a few times in quick succession, may read slightly easier to change the third to `that helps them to build`?

**Comment by jamesmcneil** [comment]
I like this as a good introduction to new starters or a reminder to us all to think more about pairing :+1:

**Comment by hpoom** [comment]
Looks good to me @steve-at-hx I would say it is weighted towards new starters, but does cover off pairing with experienced team members too. I would probably expand out the "Problem solving" chapter, but I or other can do that on a future PR. 

In order to get this PR closed and this information out there I would say make any minor tweeks you feel are needed then ShipIt :shipit: 

It also needs to be linked back into Readme.md or another file. So there should be 2 files changed on this PR, the new file on Pairing and a file it is linked in.

**Pull request merged**